### ISSUE TITLE
Updating $PATH early on doesn't affect later `zsh` calls

### ISSUE THREAD

**Issue by Techcable** [Techcable/zsh2xonsh#2]
Early in my `.xonshrc` I run `eval $(brew shellenv)` by doing
````xonsh
zsh2xonsh.translate_to_xonsh_and_eval($(/opt/homebrew/bin/brew shellenv))
````

Later on, I try to run my `~/.config.zsh`

Currently `~/.config.zsh` [invokes the `fd` command](https://github.com/Techcable/dotfiles/blob/7f087f2fba3baf8f50a4ada8bb3057958618d414/macbook-config.zsh#L7) to detect the best $JAVA_HOME.

In the config,
````shell
local preferred_java_version=17
local preferred_java_home=$(fd "jdk-${prefered_java_version}.*\.jdk" /Library/Java/JavaVirtualMachines --maxdepth 1)
````

is translated into
````xonsh
preferred_java_version=17
preferred_java_home=runtime.zsh('fd "jdk-${prefered_java_version}.*\\.jdk" /Library/Java/JavaVirtualMachines --maxdepth 1')`
````
Unfortunately, `fd` is only available from homebrew.

Based on my debugging, it appears the initial `$(brew shellenv)` does not affect the $PATH in `runtime.zsh_expand_quote()`. That is, `os.environ()` does not see the correct path.

The ultimate reason for this is that `$UPDATE_OS_ENVIRON = False` [by default](https://xon.sh/envvars.html#update-os-environ).

We need to find a way to work around this (or convince xonsh to change the defaults). Why was it set this way?

**Comment by Techcable** [comment]
EDIT: Adding `$UPDATE_OS_ENVIRON = True` doesn't seem to fix it.

Removing the `env=<>` specification from `subprocess.run ` (and defeaulting to inheriting env) doesn't work either.

Using `os.getenv('PATH')` doesn't work either.

However, if you run `env zsh -c 'echo $PATH'` or `env` directly in xonsh, then things work correctly. Even `python3 -c 'import os; print(os.environ['PATH'])'` works. The value of `os.environ()` is only incorrect if you run in the same python process, not in a subprocess :(

Right now the `runtime` module doesn't depend on `xonsh` directly. Maybe we should change that to properly detect `$PATH` (and other environment variables)?

**Comment by Techcable** [comment]
Fixed by 26094122bf06277c1a2a6e4447cbc6fb23d31b91
